lexicon/providers/wedos.py

```diff
@@ -0,0 +1,216 @@
+    "wedos.net",
+    "wedos.eu",
+    "wedos.cz",
+    "wedos.com"]
+
+
+def provider_parser(subparser):
+    """Return the parser for this provider"""
+    subparser.description = """
+    """
+    subparser.add_argument(
+        "--auth-username",
+        help="specify email address for authentication",
+    )
+    subparser.add_argument(
+        "--auth-pass",
+        help="specify password for WAPI",
+    )
+
+
+def _filter_rtype(rtype: str, rec) -> bool:
+    if rec["type"] == rtype:
+        return True
+    return False
+
+
+def _filter_name(name: str, rec) -> bool:
+    if rec["name"] == name:
+        return True
+    return False
+
+
+def _filter_content(content: str, rec) -> bool:
+    if rec["content"] == content:
+        return True
+    return False
+
+
+class Provider(BaseProvider):
+    def __init__(self, config):
+        super(Provider, self).__init__(config)
+        self.domain_id = None
+        self.api_endpoint = "https://api.wedos.com/wapi/json"
+
+    @staticmethod
+    def _auth_hash(login, password):
+        passhash = hashlib.sha1(password.encode('utf8')).hexdigest()
+        phrase = login + passhash + datetime.datetime.now().strftime('%H')
+        return hashlib.sha1(phrase.encode('utf8')).hexdigest()
+
+    def _authenticate(self):
+
+        payload = self._post(data=self._create_payload('dns-domains-list', ''))
+        domains = payload["response"]["data"]["domain"]
+        for record in domains:
+            if record["name"] == self.domain:
+                self.domain_id = record["name"]
+                break
+            else:
+                raise AuthenticationError("No domain found")
+
+        self.domain_id = self.domain
+
+    def _create_payload(self, command, payload_data):
+        data = {'request': {'user': self._get_provider_option("auth_username"),
+                            'auth': self._auth_hash(self._get_provider_option("auth_username"),
+                                                    self._get_provider_option("auth_pass")),
+                            'command': command,
+                            'data': payload_data
+                            }}
+        return {'request': json.dumps(data)}
+
+    def _create_record(self, rtype: str, name: str, content: str) -> bool:
+        records = self._list_records(rtype, name, content)
+        if len(records) == 1:
+            return True
+        data = {
+            'type': rtype,
+            'name': self._full_name(name),
+            'rdata': content,
+            'domain': self.domain_id
+        }
+        if self._get_lexicon_option("ttl"):
+            data["ttl"] = self._get_lexicon_option("ttl")
+
+        payload = self._post(data=self._create_payload('dns-row-add', data))
+        code = payload["response"]["code"]
+        if code == 1000:
+            validation = self._validate_changes()
+            if validation:
+                return True
+            else:
+                return False
```

Comment:
I put my comment on the first occurrence of this return of `False`, but it is valid also for the other occurrences for list, update and delete.

Returning `False` may not be the best thing to do when something wrong happened during the registration of a new record. Indeed there is no strong behavior associated to this (eg. non zero exit code, or overall exception raised by the client). It is just a boolean returned in the output.

It seems to me that here your provider gives the capability to initiate a post validation of the record, and that failing the validation is a real issue (bad input, duplicates etc.).

So instead I would recommend you to create a subclass of the `LexiconError` exception, specific to your provider, like `PostValidationError` and raise it with relevant information given by the API response. This way it becomes a proper failure with non zero exit code on the cli, and it is a catcheable exception for integrators. 

Wh do you think of that?

Reply:
Well, maybe the naming of the _validate_changes is not a good choice.
It is this command: https://kb.wedos.com/en/wapi-api-interface/wapi-command-dns-domain-commit/
It is some confirmation that all changes are done and it should be distributed to the DNS.
Without this command, the changes will be applied after 5 min.

Reply:
But I will introduce the LexiconError subclass and use it when whatever create/update/delete fails

Reply:
Or just throwing LexiconError will be enough?

Reply:
I changed it to raising LexiconError. 
If it will be not enough ping me.